exchanges/stream/websocket.go

```diff
@@ -285,20 +277,48 @@ func (w *Websocket) Connect() error {
 	w.trafficMonitor()
 	w.setState(connecting)
 
-	err := w.connector()
-	if err != nil {
-		w.setState(disconnected)
-		return fmt.Errorf("%v Error connecting %w", w.exchangeName, err)
+	if w.connector != nil {
+		err := w.connector()
+		if err != nil {
+			w.setState(disconnected)
+			return fmt.Errorf("%v Error connecting %w", w.exchangeName, err)
+		}
+	}
+
+	if w.Conn != nil {
```

Comment:
This setup doesn't work. Non-new websockets will panic here as w.Conn is defined, but the functions below are not

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x104adf5a8]

goroutine 120 [running]:
github.com/thrasher-corp/gocryptotrader/exchanges/stream.(*Websocket).listen(0x140001ae3c0, {0x1058830d8, 0x140008e0100}, 0x0)
        /Users/scottg/go/src/github.com/thrasher-corp/gocryptotrader/exchanges/stream/websocket.go:992 +0x78
created by github.com/thrasher-corp/gocryptotrader/exchanges/stream.(*Websocket).InitialiseConnection in goroutine 146
        /Users/scottg/go/src/github.com/thrasher-corp/gocryptotrader/exchanges/stream/websocket.go:1022 +0x1a8
```

This is for Kraken, it shouldn'tbe hitting this new code for Kraken, right? No errors are thrown from nil functions in `InistialiseConnection`